src/frontend/src/components/modals/LicenseModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans, t } from '@lingui/macro';
+import {
+  Accordion,
+  Alert,
+  Divider,
+  Group,
+  LoadingOverlay,
+  Space,
+  Stack,
+  Tabs,
+  Text
+} from '@mantine/core';
+import { ContextModalProps } from '@mantine/modals';
+import { useQuery } from '@tanstack/react-query';
+
+import { api } from '../../App';
+import { ApiEndpoints } from '../../enums/ApiEndpoints';
+import { apiUrl } from '../../states/ApiState';
+import { StylishText } from '../items/StylishText';
+
+export function LicenceView(entries: any[]) {
+  return (
+    <Stack spacing="xs">
+      <Divider />
+      {entries?.length && (
+        <Accordion variant="contained" defaultValue="-">
+          {entries?.map((entry: any, index: number) => (
+            <Accordion.Item key={index} value={`entry-${index}`}>
+              <Accordion.Control>
+                <Group position="apart" grow>
+                  <Text>{entry.name}</Text>
+                  <Text>{entry.license}</Text>
+                  <Space />
+                  <Text>{entry.version}</Text>
+                </Group>
+              </Accordion.Control>
+              <Accordion.Panel>
+                {entry.licensetext || t`No license text available`}
+              </Accordion.Panel>
+            </Accordion.Item>
+          ))}
+        </Accordion>
+      )}
+    </Stack>
+  );
+}
+
+export function LicenseModal({}: ContextModalProps<{
+  modalBody: string;
+}>) {
+  const { data, isFetching, isError } = useQuery({
+    queryKey: ['license'],
+    queryFn: () =>
+      api
+        .get(apiUrl(ApiEndpoints.license))
+        .then((res) => res.data ?? {})
+        .catch(() => {})
+  });
+
+  return (
+    <Stack spacing="xs">
+      <Divider />
+      <LoadingOverlay visible={isFetching} />
+      {isFetching && (
+        <Text>
+          <Trans>Loading license information</Trans>
+        </Text>
+      )}
+      {isError ? (
+        <Alert color="red" title={t`Error`}>
+          <Text>
+            <Trans>Failed to fetch license information</Trans>
+          </Text>
+        </Alert>
+      ) : (
+        <Tabs defaultValue="backend">
+          <Tabs.List>
+            <Tabs.Tab value="backend">
+              <Trans>Backend Packages</Trans>
+            </Tabs.Tab>
+            <Tabs.Tab value="frontend">
+              <Trans>Frontend Packages</Trans>
+            </Tabs.Tab>
+          </Tabs.List>
+
+          <Tabs.Panel value="backend">
+            {LicenceView(data?.backend ?? [])}
+          </Tabs.Panel>
+
+          <Tabs.Panel value="frontend">
+            {LicenceView(data?.frontend ?? [])}
+          </Tabs.Panel>
+        </Tabs>
```

Comment:
I would much prefer if we would not hard-code the sections here as many deployers will need to add additional dependency sections for their infra

Reply:
Would you like to make this extensibility change now, or merge as is with future work to make it extensible? This would require backend API changes too but not much

Reply:
My initial PR just had support in the frontend, that should be enough. You can always monkey-patch the backend with a plugin/custom settings.py; I will amend this PR today